### PR TITLE
[BugFix] Fix unified connector not showing iceberg table

### PR DESCRIPTION
**Pull request by letian-jiang**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedMetadata.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public UnifiedMetadata(Map<Table.TableType, ConnectorMetadata> metadataMap) {
 
     private Table.TableType getTableType(String dbName, String tblName) {
         Table table = hiveMetadata.getTable(dbName, tblName);
-        if (table == null) {
+        if (table == null || table.isHiveView()) {
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedMetadata.java:74]: To support hive view, redirect to hive connector

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/RemoteFileInputFormat.java [hunk 2]
```diff
@@ -45,11 +44,7 @@ public enum RemoteFileInputFormat {
                     .build();
 
     public static RemoteFileInputFormat fromHdfsInputFormatClass(String className) {
-        RemoteFileInputFormat fileInputFormat = VALID_INPUT_FORMATS.get(className);
-        if (fileInputFormat == null) {
-            throw new StarRocksConnectorException("Unsupported file format: %s", className);
-        }
-        return fileInputFormat;
+        return VALID_INPUT_FORMATS.getOrDefault(className, UNKNOWN);
```
Comment (letian-jiang) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/RemoteFileInputFormat.java:47]: return `UNKNOWN` for non-hive supported formats

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/RemoteFileInputFormat.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/unified/UnifiedMetadata.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/RemoteFileInputFormat.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/unified/UnifiedMetadata.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/RemoteFileInputFormat.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/unified/UnifiedMetadata.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**